---
title: "ggmsa: Plot multiple sequence alignment using ggplot2"
output: 
  prettydoc::html_pretty:
    toc: true
    theme: hpstr
date: "`r Sys.Date()`"
bibliography: ggmsa.bib
vignette: >
  %\VignetteIndexEntry{ggmsa}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

CRANpkg <- function(pkg) {
    cran <- "https://cran.r-project.org/package"
    fmt <- "[%s](%s=%s)"
    sprintf(fmt, pkg, cran, pkg)
}

Biocpkg <- function(pkg) {
    sprintf("[%s](http://bioconductor.org/packages/%s)", pkg, pkg)
}

library(ggmsa)
library(ggplot2)
```


# Abstract

Plot multiple sequence alignment using `r CRANpkg("ggplot2")` with multiple color schemes supported.

# Citation

To do

# Introduction

Supports visualizing multiple sequence alignment of DNA and protein sequences using ggplot2 It supports a number of colour schemes, including Chemistry, Clustal, Shapely, Taylor and Zappo. Multiple sequence alignment can easily be combined with other 'ggplot2' plots, such as aligning a phylogenetic tree produced by 'ggtree' with multiple sequence alignment.

# Installation

This R package (`r CRANpkg("ggmsa")`, current version: `r packageVersion('ggmsa')`) is avalable via CRAN. The way to install the package is the following:

```{r eval=FALSE}
## installing the package
install.packages("ggmsa")
## loading the package
library("ggmsa")
```

## Load sample data

Three sample data are shipped with the ggmsa package. Note that ggmsa supports not only fasta files but other objects as well. `available_msa()`can be used to list MSA objects currently available.

```{r warning=FALSE}
 available_msa()

 protein_sequences <- system.file("extdata", "sample.fasta", package = "ggmsa")
 miRNA_sequences <- system.file("extdata", "seedSample.fa", package = "ggmsa")
 nt_sequences <- system.file("extdata", "LeaderRepeat_All.fa", package = "ggmsa")
 
```

# Visualizing Multiple Sequence Alignments

The most simple code to use ggmsa:
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(miRNA_sequences)
```

##  Colour Schemes
Several classic color schemes for MSA are shipped in the package. In the same ways, you can use `available_msa()` to list  colour schemes currently available. Note that Amino acids and nucleotides have different colour schemes.
```{r warning=FALSE}
 available_colors()
```

### Clustal X Colour Scheme(Default)

This is an emulation of the default colourscheme used for alignments in [Clustal X](http://www.jalview.org/help/html/colourSchemes/clustal.html), a graphical interface for the ClustalW multiple sequence alignment program. Each residue in the alignment is assigned a colour if the amino acid profile of the alignment at that position meets some minimum criteria specific for the residue type.

```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, color = "Clustal")
```

### Color by Chemistry

Amino acids are colored according to their side chain chemistry:

```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, color = "Chemistry_AA")
```

### Color by Shapely

This color scheme matches the RasMol amino acid and RasMol nucleotide [color schemes](http://openrasmol.org/doc/rasmol.html#shapelycolours), which are, in turn, based on Robert Fletterickâ€™s "Shapely models".

```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, color = "Shapely_AA")
```

### Color by Taylor

This color scheme is taken from Taylor[@Taylor1997Residual] and is also used in JalView[@Waterhouse2009Jalview].

```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, color = "Taylor_AA")
```

### Color by Zappo

This scheme colors residues according to their physico-chemical properties, and is also used in JalView[@Waterhouse2009Jalview].

```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, color = "Zappo_AA")
```

## Font

Several classic font for MSA are shipped in the package. In the same ways, you can use `available_fonts()` to list  font currently available

```{r warning=FALSE}
 available_fonts()
```

```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, font = "helvetical", color = "Chemistry_AA")
```
```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, font = "TimesNewRoman", color = "Chemistry_AA")
```
```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, font = "DroidSansMono", color = "Chemistry_AA")
```

If you specify `font = NULL`, only tiles will be plot.
```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, font = NULL, color = "Chemistry_AA")
```


## Other params

**Characters width** can be specified by `char_width`. Defaults is 0.9.
```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, char_width = 0.5, color = "Chemistry_AA")
```


**Background** can be specified by `none_bg`. If `none_bg = TRUE`, only the character will be plot.
```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, start = 320, end = 360, none_bg = TRUE, color = "Chemistry_AA")
```

**Position Highligthed** can be specified by `posHighligthed`. The `none_bg = FALSE` when you specified position Highligthed by  `posHighligthed`
```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, 164, 213, color = "Chemistry_AA", 
      posHighligthed = c(185, 190))
```

**Sequence names**  Defaults is 'NULL' which indicates that the sequence name is displayed when `font = NULL`, but 'font = char' will not be displayed.


If `seq_name = TRUE` the sequence name will be displayed when you need it. 
```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, 164, 213, color = "Chemistry_AA", seq_name = TRUE)
```


If `seq_name = FALSE` the sequence name will not be displayed in any case.
```{r fig.height = 3, fig.width = 10, warning=FALSE}
ggmsa(protein_sequences, 164, 213, font = NULL, color = "Chemistry_AA", seq_name = FALSE)
```


# Related Extensions of ggmsa

ggmsa is based on ggplot2, so combining multiple sequence alignment with other plots generated by ggplot2 is available.

## Combining with tree `ggtree::facet_msa()`

Cross-link with `r Biocpkg("ggtree")`[@yu2017ggtree] package:

```{r fig.height=2, fig.width=10, message=FALSE, warning=FALSE}
library(Biostrings)
x <- readAAStringSet(protein_sequences)
d <- as.dist(stringDist(x, method = "hamming")/width(x)[1])
library(ape)
tree <- bionj(d)
library(ggtree)
p <- ggtree(tree ) + geom_tiplab()

data <- tidy_msa(x, 164, 213)
p + geom_facet(geom = geom_msa, data = data,  panel = 'msa',
               font = NULL, color = "Chemistry_AA") +
    xlim_tree(1)
```

## Plot sequence motifs `geom_seqlogo()` 

Visualizing multiple sequence alignment with sequence logo.

```{r fig.height=6, fig.width=13, message=FALSE, warning=FALSE}
f <- system.file("extdata", "LeaderRepeat_All.fa", package = "ggmsa")
ggmsa(nt_sequences, font = NULL, color = "Chemistry_NT" ) + geom_seqlogo()
```

## Plot GC content `geom_GC()`

 Visualizing multiple sequence alignment with GC content.

```{r fig.height=6, fig.width=13, message=FALSE, warning=FALSE}
ggmsa(nt_sequences, font = NULL, color = "Chemistry_NT" ) + geom_GC()
```

## Highlighting the seed in miRNA sequences `geom_seed()`

We provide two methods for highlighting seed sequences.

+ The seed region is shaded

```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(miRNA_sequences, font = 'DroidSansMono', color = "Chemistry_NT", none_bg = TRUE) + 
   geom_seed(seed = "GAGGUAG") + theme_void()
```

+ The seed region is highlighted using star

```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(miRNA_sequences, font = 'DroidSansMono', color = "Chemistry_NT") + 
   geom_seed(seed = "GAGGUAG", star = TRUE) + theme_void()
```

## Breaking down MSA `facet_msa()`

```{r fig.height = 4, fig.width = 10, warning=FALSE}
 # 3 fields
 ggmsa(protein_sequences, end = 300, font = NULL, color = "Chemistry_AA") + facet_msa(field = 100)
```

```{r fig.height = 5, fig.width = 10, warning=FALSE}
 # 4 fields 
 ggmsa(protein_sequences, end = 400, color = "Chemistry_AA") + facet_msa(field = 100)
```
# References
