---
title: "ggmsa: Plot multiple sequence alignment with multiple color schemes supported"
output: 
  prettydoc::html_pretty:
    toc: true
    theme: hpstr
date: "`r Sys.Date()`"
bibliography: ggmsa.bib
vignette: >
  %\VignetteIndexEntry{ggmsa}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

CRANpkg <- function(pkg) {
    cran <- "https://CRAN.R-project.org/package"
    fmt <- "[%s](%s=%s)"
    sprintf(fmt, pkg, cran, pkg)
}

Biocpkg <- function(pkg) {
    sprintf("[%s](http://bioconductor.org/packages/%s)", pkg, pkg)
}
```

```{r echo=FALSE, results='hide', message=FALSE}
library(ggmsa)
```

#Abstract
Plot multiple sequence alignment using `r CRANpkg("ggpolt2")` with multiple color schemes supported.

#Citation
To do

#Introduction
Supports visualizing multiple sequence alignment of DNA and protein sequences using ggpolt2 It supports a number of colour schemes, including Chemistry, Clustal, Shapely, Taylor and Zappo. Multiple sequence alignment can easily be combined with other 'ggplot2' plots, such as aligning a phylogenetic tree produced by 'ggtree' with multiple sequence alignment.

#Installation
This R package (`r CRANpkg("ggmsa")`, current version: 0.0.1) is avalable via CRAN. The way to install the package is the following:

```{r }
## installing the package
install.packages("ggmsa")
## ## loading the package
library("ggmsa")
```

##Load sample data
The file sample.fasta is shipped with the ggmsa package. To determine where the file is located, enter the following command in your R session:
```{r warning=FALSE}
sequences <- system.file("extdata/sample.fasta", package = "ggmsa")
print(sequences)
```

#Printing Multiple Sequence Alignments
We offer 5 color schemes for multiple sequence alignments.

##Clustal X Colour Scheme
This is an emulation of the default colourscheme used for alignments in [Clustal X](http://www.jalview.org/help/html/colourSchemes/clustal.html), a graphical interface for the ClustalW multiple sequence alignment program. Each residue in the alignment is assigned a colour if the amino acid profile of the alignment at that position meets some minimum criteria specific for the residue type.
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Clustal")
```

##Color by Chemistry
Amino acids are colored according to their side chain chemistry:
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Chemistry_AA")
```

##Color by Shapely
This color scheme matches the RasMol amino acid and RasMol nucleotide [color schemes](http://openrasmol.org/doc/rasmol.html#shapelycolours), which are, in turn, based on Robert Fletterickâ€™s "Shapely models".
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Shapely_AA")
```

##Color by Taylor
This color scheme is taken from Taylor[@Taylor1997Residual] and is also used in JalView[@Waterhouse2009Jalview].
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Taylor_AA")
```

##Color by Zappo
This scheme colors residues according to their physico-chemical properties, and is also used in JalView[@Waterhouse2009Jalview].
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Zappo_AA")
```

##Do not print letters
If you specify font = NULL, only the background box will be printed.
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, font = NULL, color = "Chemistry_AA")
```

#Combining plots
ggmsa is based on ggplot2, so combining multiple sequence alignment with other plots generated by ggplot2 is simple

##Combining with tree
Cross-link with `r Biocpkg("ggtree")`[@yu2017ggtree] package:
```{r fig.height=2, fig.width=10, message=FALSE, warning=FALSE}
library(Biostrings)
x <- readAAStringSet(sequences)
d <- as.dist(stringDist(x, method = "hamming")/width(x)[1])
library(ape)
tree <- bionj(d)
library(ggtree)
p <- ggtree(tree ) + geom_tiplab()

data = ggmsa:::msa_data(sequences, 164, 213, font = NULL, color = "Chemistry_AA")
p + geom_facet(geom = ggmsa:::geom_msa, data = data,  panel = 'msa') + xlim_tree(1)
```

##Combining with sequence logo
Cross-link with `r CRANpkg("ggseqlogo")`[@Wagih2017ggseqlogo] package. Bits of sequence letters can be indicated in the sequence logo. 
```{r fig.height=7, fig.width=10, message=FALSE, warning=FALSE}
#nt <- system.file("extdata//LeaderRepeat_All.fa", package = "ggmsa")
fasta_input <- read.table("/home/zhoulang/github/YuLab_SMU/ggmsa/inst/extdata/LeaderRepeat_All.fa", header = F, row.names = NULL)
#fasta_input <- read.table(nt, header = F, row.names = NULL)
fasta_input <- as.vector(fasta_input$V1)
fasta_tidy <- fasta_input[!grepl('>', fasta_input)]
p1 <- ggmsa("/home/zhoulang/github/YuLab_SMU/ggmsa/inst/extdata/LeaderRepeat_All.fa", font = NULL, color = 'Chemistry_NT')
#p1 <- ggmsa(nt, font = NULL, color = 'Chemistry_NT')
library(ggseqlogo)
p2 <- ggplot() + geom_logo(fasta_tidy) + ggplot2::coord_fixed(2.5) 
library(cowplot)
plot_grid(p1, p2, ncol = 1, align = 'v', rel_heights = c(2.8, 1))
```
#References
