---
title: "ggmsa: Plot multiple sequence alignment with multiple color schemes supported"
output: 
  prettydoc::html_pretty:
    toc: true
    theme: hpstr
date: "`r Sys.Date()`"
vignette: >
  %\VignetteIndexEntry{ggmsa}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r echo=FALSE, results='hide', message=FALSE}
library(ggmsa)
```

# Abstract
Plot multiple sequence alignment using ggplot2 with multiple color schemes supported.

#Citation
To do

#Introduction
Supports visualizing multiple sequence alignment of DNA and protein sequences using 'ggplot2'. It supports a number of colour schemes, including Chemistry, Clustal, Shapely, Taylor and Zappo. Multiple sequence alignment can easily be combined with other 'ggplot2' plots, such as aligning a phylogenetic tree produced by 'ggtree' with multiple sequence alignment.

#Installation
This R package ("ggmsa", current version: 0.0.1) is avalable via CRAN. The way to install the package is the following:

```{r }
## installing the package
install.packages("ggmsa")
## ## loading the package
library("ggmsa")
```

##Setting up the data
The file sample.fasta is shipped with the msa package. To determine where the file is located, enter the following command in your R session:
```{r warning=FALSE}
sequences <- system.file("extdata/sample.fasta", package = "ggmsa")
print(sequences)
```

#Printing Multiple Sequence Alignments
We offer 5 color schemes for multiple sequence alignments.

##Clustal X Colour Scheme
This is an emulation of the default colourscheme used for alignments in Clustal X, a graphical interface for the ClustalW multiple sequence alignment program. Each residue in the alignment is assigned a colour if the amino acid profile of the alignment at that position meets some minimum criteria specific for the residue type.
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Clustal")
```

##Color by Chemistry
Amino acids are colored according to their side chain chemistry:
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Chemistry_AA")
```

##Shapely
This color scheme matches the RasMol amino acid and RasMol nucleotide color schemes, which are, in turn, based on Robert Fletterickâ€™s "Shapely models".
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Shapely_AA")
```

##Taylor
This color scheme is taken from Taylor and is also used in JalView.
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Taylor_AA")
```

##Zappo
This scheme colors residues according to their physico-chemical properties, and is also used in JalView.
```{r fig.height = 2, fig.width = 10, warning=FALSE}
ggmsa(sequences, 320, 360, color = "Zappo_AA")
```

#Cross-link with other packages
```{r fig.height=2, fig.width=10, message=FALSE, warning=FALSE}
library(Biostrings)
x <- readAAStringSet(sequences)
d <- as.dist(stringDist(x, method = "hamming")/width(x)[1])
library(ape)
tree <- bionj(d)
library(ggtree)
p <- ggtree(tree ) + geom_tiplab()

data = ggmsa:::msa_data(sequences, 164, 213, color = "Chemistry_AA")
p + geom_facet(geom = ggmsa:::geom_msa, data = data,  panel = 'msa') + xlim_tree(1)
```

#References
Todo