<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ggmsa: Plot multiple sequence alignment using ggplot2 • ggmsa</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="ggmsa: Plot multiple sequence alignment using ggplot2">
<meta property="og:description" content="ggmsa">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggmsa</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/ggmsa.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Extensions/extensions.html"> Extensions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>ggmsa: Plot multiple sequence alignment using ggplot2</h1>
            
            <h4 class="date">2020-05-25</h4>
      
      
      <div class="hidden name"><code>ggmsa.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Plot multiple sequence alignment using <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> with multiple color schemes supported.</p>
<p>Supports visualizing multiple sequence alignment of DNA and protein sequences using ggplot2 It supports a number of colour schemes, including Chemistry, Clustal, Shapely, Taylor and Zappo. Multiple sequence alignment can easily be combined with other ‘ggplot2’ plots, such as aligning a phylogenetic tree produced by ‘ggtree’ with multiple sequence alignment.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>This R package (<a href="https://cran.r-project.org/package=ggmsa">ggmsa</a>, current version: 0.0.4) is avalable via CRAN. The way to install the package is the following:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co">## installing the package</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"ggmsa"</span>)
<span class="co">## loading the package</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"ggmsa"</span>)</pre></body></html></div>
<div id="load-sample-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-sample-data" class="anchor"></a>Load sample data</h2>
<p>Three sample data are shipped with the ggmsa package. Note that ggmsa supports not only fasta files but other objects as well. <code><a href="../reference/available_msa.html">available_msa()</a></code>can be used to list MSA objects currently available.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"> <span class="fu"><a href="../reference/available_msa.html">available_msa</a></span>()
<span class="co">#&gt; files currently available:</span>
<span class="co">#&gt; .fasta</span>
<span class="co">#&gt; XStringSet objects from 'Biostrings' package:</span>
<span class="co">#&gt; DNAStringSet RNAStringSet AAStringSet BStringSet DNAMultipleAlignment RNAMultipleAlignment AAMultipleAlignment</span>
<span class="co">#&gt; bin objects from 'seqmagick' package:</span>
<span class="co">#&gt; DNAbin AAbin</span>

 <span class="no">protein_sequences</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"sample.fasta"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"ggmsa"</span>)
 <span class="no">miRNA_sequences</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"seedSample.fa"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"ggmsa"</span>)
 <span class="no">nt_sequences</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"LeaderRepeat_All.fa"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"ggmsa"</span>)</pre></body></html></div>
</div>
</div>
<div id="visualizing-multiple-sequence-alignments" class="section level1">
<h1 class="hasAnchor">
<a href="#visualizing-multiple-sequence-alignments" class="anchor"></a>Visualizing Multiple Sequence Alignments</h1>
<p>The most simple code to use ggmsa:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">265</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">300</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-4-1.png" width="960"></p>
<div id="colour-schemes" class="section level2">
<h2 class="hasAnchor">
<a href="#colour-schemes" class="anchor"></a>Colour Schemes</h2>
<p>Several classic color schemes for MSA are shipped in the package. In the same ways, you can use <code><a href="../reference/available_msa.html">available_msa()</a></code> to list colour schemes currently available. Note that Amino acids and nucleotides have different colour schemes.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"> <span class="fu"><a href="../reference/available_colors.html">available_colors</a></span>()
<span class="co">#&gt; color schemes for nucleotide sequences currently available:</span>
<span class="co">#&gt; Chemistry_NT Shapely_NT Taylor_NT Zappo_NT</span>
<span class="co">#&gt; color schemes for AA sequences currently available:</span>
<span class="co">#&gt; Clustal Chemistry_AA Shapely_AA Zappo_AA Taylor_AA</span></pre></body></html></div>
<div id="clustal-x-colour-schemedefault" class="section level3">
<h3 class="hasAnchor">
<a href="#clustal-x-colour-schemedefault" class="anchor"></a>Clustal X Colour Scheme(Default)</h3>
<p>This is an emulation of the default colourscheme used for alignments in <a href="http://www.jalview.org/help/html/colourSchemes/clustal.html">Clustal X</a>, a graphical interface for the ClustalW multiple sequence alignment program. Each residue in the alignment is assigned a colour if the amino acid profile of the alignment at that position meets some minimum criteria specific for the residue type.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Clustal"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-6-1.png" width="960"></p>
</div>
<div id="color-by-chemistry" class="section level3">
<h3 class="hasAnchor">
<a href="#color-by-chemistry" class="anchor"></a>Color by Chemistry</h3>
<p>Amino acids are colored according to their side chain chemistry:</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
</div>
<div id="color-by-shapely" class="section level3">
<h3 class="hasAnchor">
<a href="#color-by-shapely" class="anchor"></a>Color by Shapely</h3>
<p>This color scheme matches the RasMol amino acid and RasMol nucleotide <a href="http://openrasmol.org/doc/rasmol.html#shapelycolours">color schemes</a>, which are, in turn, based on Robert Fletterick’s “Shapely models”.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Shapely_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-8-1.png" width="960"></p>
</div>
<div id="color-by-taylor" class="section level3">
<h3 class="hasAnchor">
<a href="#color-by-taylor" class="anchor"></a>Color by Taylor</h3>
<p>This color scheme is taken from Taylor<span class="citation">(Taylor 1997)</span> and is also used in JalView<span class="citation">(Waterhouse et al. 2009)</span>.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Taylor_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-9-1.png" width="960"></p>
</div>
<div id="color-by-zappo" class="section level3">
<h3 class="hasAnchor">
<a href="#color-by-zappo" class="anchor"></a>Color by Zappo</h3>
<p>This scheme colors residues according to their physico-chemical properties, and is also used in JalView<span class="citation">(Waterhouse et al. 2009)</span>.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Zappo_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-10-1.png" width="960"></p>
</div>
</div>
<div id="font" class="section level2">
<h2 class="hasAnchor">
<a href="#font" class="anchor"></a>Font</h2>
<p>Several classic font for MSA are shipped in the package. In the same ways, you can use <code><a href="../reference/available_fonts.html">available_fonts()</a></code> to list font currently available</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"> <span class="fu"><a href="../reference/available_fonts.html">available_fonts</a></span>()
<span class="co">#&gt; font families currently available:</span>
<span class="co">#&gt; helvetical mono TimesNewRoman DroidSansMono</span></pre></body></html></div>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">font</span> <span class="kw">=</span> <span class="st">"helvetical"</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-12-1.png" width="960"></p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">font</span> <span class="kw">=</span> <span class="st">"TimesNewRoman"</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-13-1.png" width="960"></p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">font</span> <span class="kw">=</span> <span class="st">"DroidSansMono"</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-14-1.png" width="960"></p>
<p>If you specify <code>font = NULL</code>, only tiles will be plot.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">font</span> <span class="kw">=</span> <span class="kw">NULL</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-15-1.png" width="960"></p>
</div>
<div id="other-params" class="section level2">
<h2 class="hasAnchor">
<a href="#other-params" class="anchor"></a>Other params</h2>
<p><strong>Characters width</strong> can be specified by <code>char_width</code>. Defaults is 0.9.</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">char_width</span> <span class="kw">=</span> <span class="fl">0.5</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-16-1.png" width="960"></p>
<p><strong>Background</strong> can be specified by <code>none_bg</code>. If <code>none_bg = TRUE</code>, only the character will be plot.</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="kw">start</span> <span class="kw">=</span> <span class="fl">320</span>, <span class="kw">end</span> <span class="kw">=</span> <span class="fl">360</span>, <span class="kw">none_bg</span> <span class="kw">=</span> <span class="fl">TRUE</span>) + <span class="fu">theme_void</span>()</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-17-1.png" width="960"></p>
<p><strong>Position Highligthed</strong> can be specified by <code>posHighligthed</code>. The <code>none_bg = FALSE</code> when you specified position Highligthed by <code>posHighligthed</code></p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="fl">164</span>, <span class="fl">213</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>,
      <span class="kw">posHighligthed</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">185</span>, <span class="fl">190</span>))</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-18-1.png" width="960"></p>
<p><strong>Sequence names</strong> Defaults is ‘NULL’ which indicates that the sequence name is displayed when <code>font = NULL</code>, but ‘font = char’ will not be displayed.</p>
<p>If <code>seq_name = TRUE</code> the sequence name will be displayed when you need it.</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="fl">164</span>, <span class="fl">213</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>, <span class="kw">seq_name</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-19-1.png" width="960"></p>
<p>If <code>seq_name = FALSE</code> the sequence name will not be displayed in any case.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span>(<span class="no">protein_sequences</span>, <span class="fl">164</span>, <span class="fl">213</span>, <span class="kw">font</span> <span class="kw">=</span> <span class="kw">NULL</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="st">"Chemistry_AA"</span>, <span class="kw">seq_name</span> <span class="kw">=</span> <span class="fl">FALSE</span>)</pre></body></html></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-20-1.png" width="960"></p>
</div>
</div>
<div id="reference" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#reference" class="anchor"></a>Reference</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-Taylor1997Residual">
<p>Taylor, W R. 1997. “Residual Colours: A Proposal for Aminochromography.” <em>Protein Eng</em> 10 (7): 743–46.</p>
</div>
<div id="ref-Waterhouse2009Jalview">
<p>Waterhouse, A. M., J. B. Procter, D. M. Martin, M Clamp, and G. J. Barton. 2009. “Jalview Version 2–a Multiple Sequence Alignment Editor and Analysis Workbench.” <em>Bioinformatics</em> 25 (9): 1189.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Guangchuang Yu, Lang Zhou.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
