<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ggmsa: Getting Started • ggmsa</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="ggmsa: Getting Started">
<meta property="og:description" content="ggmsa">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggmsa</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/ggmsa.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/guides/Annotations.html">Annotation</a>
    </li>
    <li>
      <a href="../articles/guides/Color_schemes_And_Font_Families.html">Color Schemes and Font Families</a>
    </li>
    <li>
      <a href="../articles/guides/MSA_theme.html">Theme</a>
    </li>
    <li>
      <a href="../articles/guides/Other_Modules.html">Other Modules</a>
    </li>
    <li>
      <a href="../articles/guides/View_modes.html">View Modes</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ggmsa_files/kePrint-0.0.1/kePrint.js"></script><link href="ggmsa_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>ggmsa: Getting Started</h1>
            
            <h4 class="date">2021-05-12</h4>
      
      
      <div class="hidden name"><code>ggmsa.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>ggmsa is a package designed to plot multiple sequence alignments using <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> .</p>
<p>This package implements functions to visualize publication-quality multiple sequence alignments (protein/DNA/RNA) in R extremely simple and powerful. It uses module design to annotate sequence alignments and allows to accept other datasets for diagrams combination.<a href="https://cran.r-project.org/package=ggmsa">ggmsa</a>, (current version: 1.0.0) is avalable via CRAN.</p>
<p>In this tutorial, we’ll work through the basics of using ggmsa.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"ggmsa"</span><span class="op">)</span></code></pre></div>
<p><img src="man/figures/workflow.png" width="50%"></p>
</div>
<div id="importing-msa-data" class="section level1">
<h1 class="hasAnchor">
<a href="#importing-msa-data" class="anchor"></a>Importing MSA data</h1>
<p>We’ll start by importing some example data to use throughout this tutorial. Expect FASTA files, some of the objects in R can also as input. <code><a href="../reference/available_msa.html">available_msa()</a></code> can be used to list MSA objects currently available.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu"><a href="../reference/available_msa.html">available_msa</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; files currently available:</span>
<span class="co">#&gt; .fasta</span>
<span class="co">#&gt; XStringSet objects from 'Biostrings' package:</span>
<span class="co">#&gt; DNAStringSet RNAStringSet AAStringSet BStringSet DNAMultipleAlignment RNAMultipleAlignment AAMultipleAlignment</span>
<span class="co">#&gt; bin objects:</span>
<span class="co">#&gt; DNAbin AAbin</span>

 <span class="va">protein_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample.fasta"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
 <span class="va">miRNA_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"seedSample.fa"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
 <span class="va">nt_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"LeaderRepeat_All.fa"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
 </code></pre></div>
</div>
<div id="basic-use-msa-visualization" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-use-msa-visualization" class="anchor"></a>Basic use: MSA Visualization</h1>
<p>The most simple code to use ggmsa:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span><span class="op">(</span><span class="va">protein_sequences</span>, <span class="fl">300</span>, <span class="fl">350</span>, color <span class="op">=</span> <span class="st">"Clustal"</span>, font <span class="op">=</span> <span class="st">"DroidSansMono"</span>, char_width <span class="op">=</span> <span class="fl">0.5</span>, seq_name <span class="op">=</span> <span class="cn">T</span> <span class="op">)</span></code></pre></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
<div id="color-schemes" class="section level2">
<h2 class="hasAnchor">
<a href="#color-schemes" class="anchor"></a>Color Schemes</h2>
<p>ggmsa predefines several color schemes for rendering MSA are shipped in the package. In the same ways, using <code><a href="../reference/available_msa.html">available_msa()</a></code> to list color schemes currently available. Note that amino acids (protein) and nucleotides (DNA/RNA) have different names.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_colors.html">available_colors</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; color schemes for nucleotide sequences currently available:</span>
<span class="co">#&gt; Chemistry_NT Shapely_NT Taylor_NT Zappo_NT</span>
<span class="co">#&gt; color schemes for AA sequences currently available:</span>
<span class="co">#&gt; Clustal Chemistry_AA Shapely_AA Zappo_AA Taylor_AA LETTER CN6 Hydrophobicity</span></code></pre></div>
<p><img src="man/figures/schemes.png" width="50%"></p>
</div>
<div id="font" class="section level2">
<h2 class="hasAnchor">
<a href="#font" class="anchor"></a>Font</h2>
<p>Several predefined fonts are shipped ggmsa. Users can use <code><a href="../reference/available_fonts.html">available_fonts()</a></code> to list the font currently available.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_fonts.html">available_fonts</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; font families currently available:</span>
<span class="co">#&gt; helvetical mono TimesNewRoman DroidSansMono</span></code></pre></div>
</div>
</div>
<div id="msa-annotation" class="section level1">
<h1 class="hasAnchor">
<a href="#msa-annotation" class="anchor"></a>MSA Annotation</h1>
<p>ggmsa supports annotations for MSA. Similar to the ggplot2, it implements annotations by <code>geom</code> and users can perform annotation with <code><a href="https://rdrr.io/r/base/Arithmetic.html">+</a></code> , like this: <code>ggmsa() + geom_*()</code>. Automatically generated annotations that containing colored labels and symbols are overlaid on MSAs to indicate potentially conserved or divergent regions.</p>
<p>For example, visualizing multiple sequence alignment with <strong>sequence logo</strong> and <strong>bar chart</strong>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span><span class="op">(</span><span class="va">protein_sequences</span>, <span class="fl">221</span>, <span class="fl">280</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span>, char_width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_seqlogo.html">geom_seqlogo</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"Chemistry_AA"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_msaBar.html">geom_msaBar</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-9-1.png" width="1056"></p>
<p>This table shows the annnotation layers supported by ggmsa as following:</p>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
Annotation modules
</th>
<th style="text-align:left;">
Type
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
geom_seqlogo()
</td>
<td style="text-align:left;">
geometric layer
</td>
<td style="text-align:left;">
automatically generated sequence logos for a MSA
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_GC()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
shows GC content with bubble chart
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_seed()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
highlights seed region on miRNA sequences
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_msaBar()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
shows sequences conservation by a bar chart
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_helix()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
depicts RNA secondary structure as arc diagrams(need extra data)
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
</div>
<div id="combination-plot" class="section level1">
<h1 class="hasAnchor">
<a href="#combination-plot" class="anchor"></a>Combination Plot</h1>
<p>Other molecular datasets can be accepted and rendered by ggmsa (i.e. gene structure diagram, phylogenetic tree diagram, gene arrow maps). ggmsa allows users to align such associated graphs in MSA plot.</p>
<div id="msa-tree" class="section level2">
<h2 class="hasAnchor">
<a href="#msa-tree" class="anchor"></a>MSA + Tree</h2>
<p>The most important part of phylogenetic analysis is the visual inspection, annotation and exploration of phylogenetic trees. This is no surprise that aligning phylogenetic tree to the MSA plot. ggtree has called to plot phylogenetic trees.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readAAStringSet</a></span><span class="op">(</span><span class="va">protein_sequences</span><span class="op">)</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/stringDist.html">stringDist</a></span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">"hamming"</span><span class="op">)</span><span class="op">/</span><span class="fu">width</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/">ape</a></span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/bionj.html">bionj</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">data</span> <span class="op">=</span> <span class="fu"><a href="../reference/tidy_msa.html">tidy_msa</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">164</span>, <span class="fl">213</span><span class="op">)</span>
<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet</a></span><span class="op">(</span>geom <span class="op">=</span> <span class="va">geom_msa</span>, data <span class="op">=</span> <span class="va">data</span>,  panel <span class="op">=</span> <span class="st">'msa'</span>,
               font <span class="op">=</span> <span class="cn">NULL</span>, color <span class="op">=</span> <span class="st">"Chemistry_AA"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/xlim_tree.html">xlim_tree</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-11-1.png" width="960"></p>
</div>
<div id="multiple-graphs-msa-tree-gene-map" class="section level2">
<h2 class="hasAnchor">
<a href="#multiple-graphs-msa-tree-gene-map" class="anchor"></a>Multiple graphs: MSA + Tree + gene map</h2>
<p>A notable example is integrated three associated panels. MSA plot, phylogenetic tree and gene map produced by ggmsa, ggtree and gggenes respectively.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#import data</span>
<span class="va">tp53_sequences</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tp53.fa"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="va">tp53_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TP53_genes.xlsx"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>

<span class="co">#tree</span>
<span class="va">tp53</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readAAStringSet</a></span><span class="op">(</span><span class="va">tp53_sequences</span><span class="op">)</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/stringDist.html">stringDist</a></span><span class="op">(</span><span class="va">tp53</span>, method <span class="op">=</span> <span class="st">"hamming"</span><span class="op">)</span><span class="op">/</span><span class="fu">width</span><span class="op">(</span><span class="va">tp53</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/bionj.html">bionj</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span>
<span class="va">p_tp53</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, branch.length <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#msa</span>
<span class="va">data_53</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tidy_msa.html">tidy_msa</a></span><span class="op">(</span><span class="va">tp53</span><span class="op">)</span>

<span class="co">#gene maps</span>
<span class="va">TP53_arrow</span> <span class="op">&lt;-</span> <span class="fu">readxl</span><span class="fu">::</span><span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html">read_xlsx</a></span><span class="op">(</span><span class="va">tp53_genes</span><span class="op">)</span>
<span class="va">TP53_arrow</span><span class="op">$</span><span class="va">direction</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">TP53_arrow</span><span class="op">[</span><span class="va">TP53_arrow</span><span class="op">$</span><span class="va">strand</span> <span class="op">==</span> <span class="st">"reverse"</span>,<span class="st">"direction"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fl">1</span>

<span class="co">#color</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span>
<span class="va">mapping</span> <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, fill <span class="op">=</span> <span class="va">gene</span>, forward <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
<span class="va">my_pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, name <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co">#tree + gene maps + msa</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ggnewscale</span><span class="op">)</span>
<span class="va">p_tp53</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet</a></span><span class="op">(</span>geom <span class="op">=</span> <span class="va">geom_msa</span>, data <span class="op">=</span> <span class="va">data_53</span>,
                     panel <span class="op">=</span> <span class="st">'msa'</span>, font <span class="op">=</span> <span class="cn">NULL</span>,
                     border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/xlim_tree.html">xlim_tree</a></span><span class="op">(</span><span class="fl">3.5</span><span class="op">)</span> <span class="op">+</span> 
          <span class="fu"><a href="https://rdrr.io/pkg/ggnewscale/man/new_scale.html">new_scale_fill</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">my_pal</span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/facet-plot.html">geom_facet</a></span><span class="op">(</span>geom <span class="op">=</span> <span class="va">geom_motif</span>,
                     mapping <span class="op">=</span> <span class="va">mapping</span>, data <span class="op">=</span> <span class="va">TP53_arrow</span>,
                     panel <span class="op">=</span> <span class="st">'genes'</span>,  on <span class="op">=</span> <span class="st">'TP53'</span>,
                     arrowhead_height <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">3</span>, <span class="st">"mm"</span><span class="op">)</span>,
                     arrowhead_width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">1</span>, <span class="st">"mm"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-12-1.png" width="1344"></p>
</div>
</div>
<div id="layout" class="section level1">
<h1 class="hasAnchor">
<a href="#layout" class="anchor"></a>Layout</h1>
<p>Different layouts allow users to display more data in a given limited space. We offer two layouts to change MSA visualization styles.</p>
<div id="broken-down-layout" class="section level2">
<h2 class="hasAnchor">
<a href="#broken-down-layout" class="anchor"></a>Broken down layout</h2>
<p>The long sequence has be broken down and displayed in several lines by <code><a href="../reference/facet_msa.html">facet_msa()</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="co"># 4 fields</span>
 <span class="fu"><a href="../reference/ggmsa.html">ggmsa</a></span><span class="op">(</span><span class="va">protein_sequences</span>, start <span class="op">=</span> <span class="fl">0</span>, end <span class="op">=</span> <span class="fl">400</span>, font <span class="op">=</span> <span class="cn">NULL</span>, color <span class="op">=</span> <span class="st">"Chemistry_AA"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="../reference/facet_msa.html">facet_msa</a></span><span class="op">(</span>field <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-13-1.png" width="960"></p>
</div>
<div id="circular-layout" class="section level2">
<h2 class="hasAnchor">
<a href="#circular-layout" class="anchor"></a>Circular layout</h2>
<p>A specific layout of the alignment can also be displayed by linking ggtreeExtra. <code>geom_fruit</code> will automatically align MSA graphs to the tree with a couple of layouts such as circular, fan andand radial.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YuLab-SMU/ggtreeExtra/">ggtreeExtra</a></span><span class="op">)</span>
<span class="va">sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sequence-link-tree.fasta"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>

<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readAAStringSet</a></span><span class="op">(</span><span class="va">sequences</span><span class="op">)</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/stringDist.html">stringDist</a></span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">"hamming"</span><span class="op">)</span><span class="op">/</span><span class="fu">width</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/bionj.html">bionj</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span>
<span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tidy_msa.html">tidy_msa</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">120</span>, <span class="fl">200</span><span class="op">)</span>

<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout <span class="op">=</span> <span class="st">'circular'</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>align <span class="op">=</span> <span class="cn">TRUE</span>, offset <span class="op">=</span> <span class="fl">0.545</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">+</span> 
        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">1.3</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtreeExtra/man/geom_fruit.html">geom_fruit</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">data</span>, geom <span class="op">=</span> <span class="va">geom_msa</span>, offset <span class="op">=</span> <span class="fl">0</span>, 
                pwidth <span class="op">=</span> <span class="fl">1.2</span>, font <span class="op">=</span> <span class="cn">NULL</span>, border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></code></pre></div>
<p><img src="ggmsa_files/figure-html/unnamed-chunk-14-1.png" width="480"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Guangchuang Yu, Lang Zhou.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
